lib/base-compiler.ts

```diff
@@ -1006,24 +1008,24 @@ export class BaseCompiler implements ICompiler {
             [pathFlag + libDownloadPath],
             this.compiler.libPath.map(path => pathFlag + path),
             toolchainLibraryPaths.map(path => pathFlag + path),
-            this.getSharedLibraryPaths(libraries).map(path => pathFlag + path),
-            this.getSharedLibraryPaths(libraries).map(path => libPathFlag + path),
+            this.getSharedLibraryPaths(libraries, dirPath).map(path => pathFlag + path),
+            this.getSharedLibraryPaths(libraries, dirPath).map(path => libPathFlag + path),
         );
     }
 
-    protected getSharedLibraryPathsAsLdLibraryPaths(libraries): string[] {
+    protected getSharedLibraryPathsAsLdLibraryPaths(libraries, dirPath: string | undefined): string[] {
```

Comment:
Could make this `dirPath?: string` and then no need to specify it at all on line 417

Reply:
not against it, will do. Mostly used this to make sure I made the correct choice everywhere